code/game/objects/items/tools/weldingtool.dm

```diff
@@ -102,6 +102,13 @@
 	qdel(src)
 
 /obj/item/weldingtool/attack(mob/living/carbon/human/H, mob/user)
+	if(isOn())
+		use(1)
+		var/turf/location = get_turf(user)
+		location.hotspot_expose(700, 50, 1)
```

Comment:
Just move this to pre_attack(). Should be fine and gets rid of the duplicate code.

Reply:
I remember there was a reason I didn't put it there, I'll see if I recall when I start shifting stuff around

Reply:
Ah, I remember now, it was because things like the pacifist trait are handled in attack(), so if I put it there it could potentially consume fuel even if you didn't actually attack, if my logic is correct

Reply:
Alright. Fair enough. Still, turn this whole thing into a proc which gets called from attack and attack_obj. Reduces the amount of duplicate code.

Reply:
that good enough?

Reply:
I will assume it is